src/main/java/org/jetbrains/appenders/JsonLayout.java

```diff
@@ -33,137 +33,111 @@
-        fieldLabels.put("@timestamp", "@timestamp");
-        fieldLabels.put("thread", "thread");
-        fieldLabels.put("@version", "@version");
-    }
+    private class LoggerField {
+        private String defaultLabel;
+        private String renderedLabel;
+        private boolean isEnabled = true;
 
-    private void updateFieldLabel(String key, String value) {
-        fieldLabels.put(key, value);
-    }
+        LoggerField(String defaultName) {
+            this.defaultLabel = defaultName;
+            this.renderedLabel = defaultName;
+        }
 
-    private String getFieldLabel(String key) {
-        return fieldLabels.get(key);
-    }
+        LoggerField(String defaultName, String renderedLabel) {
+            this.defaultLabel = defaultName;
+            this.renderedLabel = renderedLabel;
+        }
 
-    private String getLocationFieldLabel(String key) {
-        return getFieldLabel("location." + key);
-    }
+        void updateRenderedLabel(String label) {
+            renderedLabel = label;
+        }
+
+        void disable() {
+            isEnabled = false;
+        }
 
-    private String getExceptionFieldLabel(String key) {
-        return getFieldLabel("exception." + key);
+        void enable() {
+            isEnabled = true;
+        }
     }
 
-    private enum LocationField {
-        CLASS("class"),
-        FILE("file"),
-        LINE("line"),
-        METHOD("method");
+    private class RenderedFieldLabels {
+        final LoggerField locationClass = loggerfield("location.class", "class");
+        final LoggerField locationFile = loggerfield("location.file", "file");
+        final LoggerField locationLine = loggerfield("location.line", "line");
+        final LoggerField locationMethod = loggerfield("location.method", "method");
+
+        final LoggerField exceptionClass = loggerfield("exception.class", "class");
+        final LoggerField exceptionMessage = loggerfield("exception.message", "message");
+        final LoggerField exceptionStacktrace = loggerfield("exception.stacktrace", "stacktrace");
+
+        final LoggerField exception = loggerfield("exception");
+        final LoggerField level = loggerfield("level");
+        final LoggerField location = loggerfield("location");
+        final LoggerField logger = loggerfield("logger");
+        final LoggerField message = loggerfield("message");
+        final LoggerField mdc = loggerfield("mdc");
+        final LoggerField ndc = loggerfield("ndc");
+        final LoggerField host = loggerfield("host");
+        final LoggerField path = loggerfield("path");
+        final LoggerField tags = loggerfield("tags");
+        final LoggerField timestamp = loggerfield("@timestamp");
+        final LoggerField thread = loggerfield("thread");
+        final LoggerField version = loggerfield("@version");
 
-        private final String val;
+        private final ArrayList<LoggerField> allFields = new ArrayList<LoggerField>();
```

Comment:
Could you please move the `allFields` line before all other LoggerFields. IFAIK, the fields are initialized from the first to the last, thus I expect it will fail with NPE.